llvm/lib/Analysis/InlineOrder.cpp

```diff
@@ -279,10 +285,92 @@ class PriorityInlineOrder : public InlineOrder<std::pair<CallBase *, int>> {
   const InlineParams &Params;
 };
 
+class TopDownInlineOrder : public InlineOrder<std::pair<CallBase *, int>> {
```

Comment:
Add a description (like in PluginAdvisor) explaining what this does and why it's useful